FEATURE  - HISTORY - VIEWMODEL EVENTS

Deivid117 opened the pull request:
NO MERGE - QUEDA UN PR PENDIENTE

- Se agregaron los eventos para ejecutar los casos de uso en el viewmodel. 
- Se agregaron los atributos al State.

[Tareas](https://trello.com/c/ox2DFZVh)

EVIDENCIA
https://drive.google.com/file/d/1DjUlndTv9SFU-BJNUVPfOSdKa1j1fhIm/view?usp=drive_link

Deivid117 committed:
FEATURE - Events ViewModel

- Se agregaron todos los eventos para usar en la ui y ejecutar los casos de uso.

- Se agregaron los atributos al State.

- Se agregó la función que muestra el error.

- Se eliminó una coma de más en CurrentExchangeRate.

DarioPapucho commented on app/src/main/java/com/varqulabs/dolarblue/core/presentation/utils/date/DateHelper.kt:
```diff
@@ -0,0 +1,45 @@
+    val datePart = date.substringBefore('T')
+    val (year, month, day) = datePart.split('-').map { it.toInt() }
+
+    // Obtenemos el nombre del mes en español
```
tienes codigo de otro PR, si tienes dependencias puedes usar este label para tomarlo en cuenta 
![image](https://github.com/user-attachments/assets/d798a5c6-9f52-487d-9430-d30c4fd12e0e)

Tambien puedes especificarlo en la descripcion del PR

![image](https://github.com/user-attachments/assets/90938f03-a52b-4762-8ced-a02fc1a6f406)

Deivid117 replied:
Lo actualizo

Deivid117 committed:
Merge branch 'develop' of https://github.com/VarquLabs/DolarBlueApp into feature/viewmodel-events

juanjosevarqu commented:
@Deivid117 Agrega Evidencia porfa en Foto o Video

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/core/domain/enums/CurrencyTab.kt:
```diff
@@ -0,0 +1,7 @@
+enum class CurrencyTab(val tabName: String, val columnName: String) {
+    BOB(tabName = "BOB", columnName = "conversion_table.pesosBob",),
+    ARG(tabName = "ARG", columnName = "conversion_table.pesosArg"),
+    USD(tabName = "USD", columnName = "conversion_table.dollar")
```
Podemos manejar un archivo de Constants.kt

Y asi hacer algo como "$conversion_table_name.$coinAbbrev"

No es bloqueante, tranca

Deivid117 replied:
vavava

DarioPapucho commented on app/build.gradle.kts:
```diff
@@ -117,6 +117,9 @@ dependencies {
     // Kotlin DateTime
     implementation(libs.kotlinx.datetime)
+
+    // Kotlin UUID
```
asegurate de tener los cambios de develop, ya que esta implementacion ya se encuentra en la rama develop

![image](https://github.com/user-attachments/assets/5de8ae2f-24ee-45cc-abbd-1bd47e597c04)

DarioPapucho commented on app/src/main/java/com/varqulabs/dolarblue/core/presentation/utils/date/DateHelper.kt:
de igual forma esto ya esta en develop

Deivid117 replied:
Se me había pasado, acabo de subirlos

Deivid117 commented on app/src/main/java/com/varqulabs/dolarblue/history/presentation/HistoryViewModel.kt:
```diff
@@ -2,23 +2,317 @@ package com.varqulabs.dolarblue.history.presentation
         when (event) {
             is OnClickDrawer -> emitOpenDrawer()
+            is Init -> executeGetConversionHistory()
+            is OnClickGetFavoriteConversions -> {
```
En un próximo pr se va a editar el query para esta función para devolver la lista de favoritos en orden

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/presentation/HistoryViewModel.kt:
```diff
@@ -2,23 +2,317 @@ package com.varqulabs.dolarblue.history.presentation
 
     override fun eventHandler(event: HistoryEvent) {
         when (event) {
             is OnClickDrawer -> emitOpenDrawer()
```
el eventHandler queda un poco extenso con logica en los {} de cada evento

Separemos en funciones aparte, buscando mejorar legibilidad

Deivid117 replied:
En lugar de esto
is OnClickGetFavoriteConversions -> {
                updateUi { copy(showFavoriteConversions = !showFavoriteConversions) }
                showConversions(uiState.value.showFavoriteConversions)
            }
            
Algo así?            
is OnClickGetFavoriteConversions -> function()

juanjosevarqu replied:
Si

Deivid117 commented:
> @Deivid117 Agrega Evidencia porfa en Foto o Video

Listo

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/presentation/HistoryViewModel.kt:
```diff
@@ -2,23 +2,317 @@ package com.varqulabs.dolarblue.history.presentation
+        }
+    }
+
+    private fun showConversions(showFavoriteConversions: Boolean) {
```
podemos renombrar por handleGetConversions

juanjosevarqu reviewed (commented):
un par de sugerencias, sobre todo busquemos legibilidad en el ViewModel

Deivid117 committed:
REFACTOR - Events ViewModel

- Se corrigió lógica de llamadas a funciones en los eventos para hacerlo más legible.

- Se renombraron funciones.

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/presentation/HistoryViewModel.kt:
```diff
@@ -2,23 +2,331 @@ package com.varqulabs.dolarblue.history.presentation
+                searchQuery = uiState.value.searchQuery,
+                currencyColumnName = uiState.value.currencyColumnName
+            )
+
```
remove space

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/presentation/HistoryViewModel.kt:
```diff
@@ -2,23 +2,331 @@ package com.varqulabs.dolarblue.history.presentation
+                isVisible = true,
+                selectedConversion = event.conversion
+            )
+
```
.

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/presentation/HistoryViewModel.kt:
```diff
@@ -2,23 +2,331 @@ package com.varqulabs.dolarblue.history.presentation
+                isVisible = false,
+                selectedConversion = null
+            )
+
```
.

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/presentation/HistoryViewModel.kt:
```diff
@@ -2,23 +2,331 @@ package com.varqulabs.dolarblue.history.presentation
+        if (searchQuery.isNotBlank() || searchQuery.isNotEmpty()) {
+            executeConversionHistorySearch(currencyColumnName, searchQuery)
+        } else {
+            executeGetConversionHistory()
```
aqui no deberia llamar a handleGetCOnverisions ?

Deivid117 replied:
En ese caso no porque se está haciendo una búsqueda global de todas las conversiones y el handle es solo para mostrar o las conversiones favoritas o todas

Deivid117 committed:
REFACTOR - Remove space

- Se quitaron espacios entre eventos del viewmodel

Deivid117 committed:
Merge branch 'develop' of https://github.com/VarquLabs/DolarBlueApp into feature/viewmodel-events

Deivid117 committed:
REFACTOR - Reformat

- Se reformateó el código de las funciones

- Se cambiaron nombres de funciones por unos más legibles.
